### PR TITLE
doc: application: add details on example-application

### PR DESCRIPTION
**Pull request by gmarull**
Mention example-application as a reference application.

### CODE REVIEW

File: doc/application/index.rst [hunk 1]
```diff
@@ -162,10 +162,24 @@ subdirectories which are not described here.
     package.
 
 
+Example Application
```
Comment (nashif) [inline doc/application/index.rst:165]: I think we should make it clear that this example application is based on a manifests, it is not your typical application that we have across the tree without modules. We do not want to have people go through the trouble of setting up something like for a basic application that does not require any of the items listed below.

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
I think this needs more discussion and infrastructure around the release process.

I am against this change unless we use example-application branches or tags whose manifests will be pinned to the release version of zephyr they were tested on.

So I'm saying for example:

- the zephyr 2.6 docs build should point to an example-application manifest that imports zephyr with `revision: v2.6.0` (or `v2.7.1`, `v2.7.2`, etc.)
- the zephyr 2.7 docs build should point to an example-application manifest that imports zephyr with `revision: v2.7.0`, etc.
- zephyr main can point to an example-application manifest that imports with `revision: main`

**Comment by gmarull** [comment]
> I think this needs more discussion and infrastructure around the release process.
> 
> I am against this change unless we use example-application branches or tags whose manifests will be pinned to the release version of zephyr they were tested on.
> 
> So I'm saying for example:
> 
> * the zephyr 2.6 docs build should point to an example-application manifest that imports zephyr with `revision: v2.6.0` (or `v2.7.1`, `v2.7.2`, etc.)
> * the zephyr 2.7 docs build should point to an example-application manifest that imports zephyr with `revision: v2.7.0`, etc.
> * zephyr main can point to an example-application manifest that imports with `revision: main`

I think it should be sufficient to make `example-application` follow Zephyr versioning and mention that in both Zephyr docs and example app README.

**Comment by gmarull** [comment]
> Can you explain what that means? Right now we don't have anything except a single branch with a manifest that points to zephyr `main` (in the example-application repository).

I mean, Zephyr docs should just link to https://github.com/zephyrproject-rtos/example-application independently of the Zephyr version. `example-application` should make explicit that it follows Zephyr versioning (I assume the idea is to make `example-application` `v2.6.0` point to Zephyr `v2.6.0`).

**Comment by mbolivar-nordic** [comment]
>  `example-application` should make explicit that it follows Zephyr versioning (I assume the idea is to make `example-application` `v2.6.0` point to Zephyr `v2.6.0`).

I think I agree, but "should" is a key word here. Right now it doesn't do anything like that, and I think we should sort that out before merging this.

File: doc/application/index.rst (hunk 1 above)
Comment (carlescufi) [reply doc/application/index.rst:165]: I agree. Maybe something like "Example standalone application" and then specifically mention west and the [T2 topology](https://docs.zephyrproject.org/latest/guides/west/workspaces.html#topologies-supported) specifically.

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: doc/application/index.rst [hunk 2]
```diff
@@ -162,11 +162,26 @@ subdirectories which are not described here.
     package.
 
 
+Example standalone application
+******************************
+
+A reference standalone application can be found in the `Example Application`_
+repository. It can be used as a reference on how to structure Zephyr based
+applications using the :ref:`T2 star topology <west-t2>`. It also demonstrates
+features commonly used in applications such as:
```
Comment (carlescufi) [inline doc/application/index.rst:171]:
```suggestion
applications using the :ref:`T2 star topology <west-t2>`. It also demonstrates the out-of-tree use of features commonly used in applications such as:
```

File: doc/application/index.rst [hunk 3]
```diff
@@ -162,11 +162,26 @@ subdirectories which are not described here.
...
+
+- Custom boards
+- Custom devicetree bindings
+- Out-of-tree drivers
```
Comment (carlescufi) [inline doc/application/index.rst:175]:
```suggestion
- Custom drivers
```

File: doc/application/index.rst [hunk 4]
```diff
@@ -162,11 +162,26 @@ subdirectories which are not described here.
...
+- Continuous Integration (CI) setup
+
 Creating an Application
 ***********************
 
 Follow these steps to create a new application directory. (Refer to
-:ref:`samples-and-demos` for existing applications provided as part of Zephyr.)
+the `Example Application`_ repository for a reference standalone application
```
Comment (carlescufi) [inline doc/application/index.rst:182]:
```suggestion
the `Example Application`_ repository for a reference standalone application in its own Git repository
```

File: doc/application/index.rst [hunk 5]
```diff
@@ -162,11 +162,26 @@ subdirectories which are not described here.
     package.
 
 
+Example standalone application
+******************************
+
+A reference standalone application can be found in the `Example Application`_
```
Comment (carlescufi) [inline doc/application/index.rst:168]:
```suggestion
A reference standalone application contained in its own Git repository can be found in the `Example Application`_
```

File: doc/application/index.rst [hunk 6]
```diff
@@ -162,11 +162,26 @@ subdirectories which are not described here.
...
+repository. It can be used as a reference on how to structure Zephyr based
```
Comment (carlescufi) [inline doc/application/index.rst:169]:
```suggestion
repository. It can be used as a reference on how to structure out-of-tree, Zephyr-based
```

**Review by carlescufi — CHANGES_REQUESTED** [review]
Much better!

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
> > `example-application` should make explicit that it follows Zephyr versioning (I assume the idea is to make `example-application` `v2.6.0` point to Zephyr `v2.6.0`).
> 
> I think I agree, but "should" is a key word here. Right now it doesn't do anything like that, and I think we should sort that out before merging this.

Agreed, maybe a quick sentence here or in the example application .md @gmarull?

**Comment by carlescufi** [comment]
@gmarull 
As part of this PR, can you also add a line in the "major enhancements" section of the [release notes](https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/release-notes-2.6.rst) with something along the lines of: "Added an example standalone Zephyr application in its own Git repository"

**Comment by gmarull** [comment]
> > > `example-application` should make explicit that it follows Zephyr versioning (I assume the idea is to make `example-application` `v2.6.0` point to Zephyr `v2.6.0`).
> > 
> > 
> > I think I agree, but "should" is a key word here. Right now it doesn't do anything like that, and I think we should sort that out before merging this.
> 
> Agreed, maybe a quick sentence here or in the example application .md @gmarull?

Will the example-application follow Zephyr versioning? I understand this has not been formally decided yet (maybe I missed it).

If you guys confirm this will be the rule I can update example-applicacion readme. We should likely include the procedure as part of the release process, too.

**Comment by gmarull** [comment]
> Added an example standalone Zephyr application in its own Git repository

done

**Comment by carlescufi** [comment]
> > > > `example-application` should make explicit that it follows Zephyr versioning (I assume the idea is to make `example-application` `v2.6.0` point to Zephyr `v2.6.0`).
> > > 
> > > 
> > > I think I agree, but "should" is a key word here. Right now it doesn't do anything like that, and I think we should sort that out before merging this.
> > 
> > 
> > Agreed, maybe a quick sentence here or in the example application .md @gmarull?
> 
> Will the example-application follow Zephyr versioning? I understand this has not been formally decided yet (maybe I missed it).

I believe we did decide that, yes. I was not able to find the relevant comment after a search, but why wouldn't we do this? This is now part of the release, and so it should be tagged all the same.

> If you guys confirm this will be the rule I can update example-applicacion readme. We should likely include the procedure as part of the release process, too.

If this PR makes it to 2.6, then let's update: https://github.com/zephyrproject-rtos/zephyr/issues/32293

**Comment by gmarull** [comment]
> I believe we did decide that, yes. I was not able to find the relevant comment after a search, but why wouldn't we do this? This is now part of the release, and so it should be tagged all the same.

I agree we should follow the same versioning, we just need to make sure it's properly documented.

**Comment by carlescufi** [comment]
> > I believe we did decide that, yes. I was not able to find the relevant comment after a search, but why wouldn't we do this? This is now part of the release, and so it should be tagged all the same.
> 
> I agree we should follow the same versioning, we just need to make sure it's properly documented.

Let's push a PR to the example-application repo stating that fact then, I do not foresee any controversy.

**Comment by gmarull** [comment]
> Let's push a PR to the example-application repo stating that fact then, I do not foresee any controversy.

https://github.com/zephyrproject-rtos/example-application/pull/9

**Review by nashif — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**